lms/views/api/canvas/files.py

```diff
@@ -44,23 +42,24 @@ def __init__(self, request):
 
     @view_config(request_method="GET", route_name="canvas_api.courses.files.list")
     def list_files(self):
-        """Return the list of files in the given course."""
-        try:
-            return self.canvas_api_client.list_files(
-                self.request.matchdict["course_id"]
-            )
-        except CanvasAPIError as err:
-            raise HTTPInternalServerError(explanation=err.explanation) from err
+        """
+        Return the list of files in the given course.
+
+        :raise lms.services.CanvasAPIError: if the Canvas API request fails.
+            This exception is caught and handled by an exception view.
+        """
+        return self.canvas_api_client.list_files(self.request.matchdict["course_id"])
```

Comment:
This turns out to be a nice pattern: views don't need to catch exceptions (in this case service-layer exceptions that `canvas_api_client` raises), instead the view just lets the exception fly and it's caught and handled by an exception view.